src/app/modules/item/pages/item-display/item-display.component.ts

```diff
@@ -0,0 +1,165 @@
+import { AfterViewInit, Component, ElementRef, Input, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { DomSanitizer, SafeResourceUrl } from '@angular/platform-browser';
+import { ItemData } from '../../services/item-datasource.service';
+import { CompleteFunction, ErrorFunction, Platform, Task, TaskParams, TaskProxyManager } from 'src/app/shared/task/task-xd-pr';
+import { interval, Subscription } from 'rxjs';
+
+interface TaskTab {
+  name: string
+}
+
+@Component({
+  selector: 'alg-item-display',
+  templateUrl: './item-display.component.html',
+  styleUrls: [ './item-display.component.scss' ]
+})
+export class ItemDisplayComponent implements OnInit, AfterViewInit, OnDestroy {
+  @Input() itemData?: ItemData;
+  @ViewChild('iframe') iframe?: ElementRef<HTMLIFrameElement>;
+
+  state : 'loading' | 'loaded' | 'unloading';
+  url? : SafeResourceUrl;
+  msg = '';
+
+  tabs: TaskTab[] = [];
+  activeTab: TaskTab;
+
+  taskProxyManager: TaskProxyManager;
+  task?: Task;
+  platform? : Platform;
+
+  height: number;
+  heightInterval? : Subscription;
+
+  lastAnswer = '';
+  lastState = '';
+  saveInterval? : Subscription;
+
+  constructor(private sanitizer: DomSanitizer) {
+    this.state = 'loading';
+    this.height = 400;
+    const initialTab = { name: 'Task' };
+    this.tabs = [ initialTab ];
+    this.tabs.push({ name: 'Editor' });
+    this.activeTab = initialTab;
+    this.taskProxyManager = new TaskProxyManager();
+  }
+
+
+  // Lifecycle functions
+  ngOnInit(): void {
+    const url = this.itemData?.item.url || '';
+    // TODO get sToken
+    const sToken = '';
+    this.setUrl(this.taskProxyManager.getUrl(url, sToken, 'http://algorea.pem.dev', 'task-'));
```

Comment:
`'http://algorea.pem.dev'` could be fetched from a configuration option at some point. It's an identifier for the platform, which needs to be given to any task supporting tokens so it can check the token against the right public key.

`'task-'` is a prefix for the randomly generated jschannel channel ID. In the original version of task-xd-pr, it was set so that we could have multiple tasks loaded through various ways, and this prefix would depend on the component displaying the task, to make sure we don't have collisions. It *could* maybe be removed.